src/main/java/com/aws/greengrass/deployment/DeploymentDocumentDownloader.java

```diff
@@ -56,16 +61,13 @@
 public class DeploymentDocumentDownloader {
     private static final Logger logger = LogManager.getLogger(DeploymentDocumentDownloader.class);
     private static final long MAX_DEPLOYMENT_DOCUMENT_SIZE_BYTES = 10 * ONE_MB;
+    private static final int MAX_CLIENT_ERROR_RETRY_COUNT = 10;
     private final GreengrassServiceClientFactory greengrassServiceClientFactory;
     private final HttpClientProvider httpClientProvider;
     private final DeviceConfiguration deviceConfiguration;
     @Setter(AccessLevel.PACKAGE)
     @Getter(AccessLevel.PACKAGE)
-    private RetryUtils.RetryConfig clientExceptionRetryConfig =
-            RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofMinutes(1))
-                    .maxRetryInterval(Duration.ofMinutes(1)).maxAttempt(Integer.MAX_VALUE)
-                    .retryableExceptions(Arrays.asList(RetryableDeploymentDocumentDownloadException.class,
-                            DeviceConfigurationException.class, RetryableServerErrorException.class)).build();
+    private RetryUtils.DifferentiatedRetryConfig clientExceptionRetryConfig;
```

Comment:
maybe the differentiated config should just be a map of list of exceptions to the retry config to use for those exceptions, where the retry config is the number of times to retry, and backoff settings.

Reply:
simplified it